src/Service/Apps/Quap/QuapService.php

```diff
@@ -183,40 +187,242 @@ public function getAnswers(Group $group, ?\DateTimeImmutable $dateTime): Answers
+     * @param AggregatedQuap $a
+     * @param AggregatedQuap $b
+     * @return int
+     */
+    private function sortByGroupType(AggregatedQuap $a, AggregatedQuap $b): int
+    {
+        return $a->getGroup()->getGroupType()->getId() - $b->getGroup()->getGroupType()->getId();
+    }
+
+    /**
+     * @param AggregatedQuap[] $quaps
+     * @return HierarchyDTO<AggregatedQuap>[]
+     */
+    private function mapToCantonalHierarchyAggregatedQuad(array $quaps): array
+    {
+        usort($quaps, fn($a, $b) => $this->sortByGroupType($a, $b));
+
+        /**
+         * maps a group_id of the canton to a HierarchyDTO of AggregatedQuap
+         * @var array<int, HierarchyDTO<AggregatedQuap>> $nestedDepartmentsMap
+         */
+        $nestedDepartmentsMap = array();
+        /**
+         * @var AggregatedQuap[] $parentlessDepartments
+         */
+        $parentlessDepartments = array();
+
+        foreach ($quaps as $quap) {
+            $group = $quap->getGroup();
+            $groupType = $group->getGroupType()->getGroupType();
+
+            if ($groupType === GroupType::CANTON) {
+                    $nestedDepartmentsMap[$group->getId()] = new HierarchyDTO($quap);
+                    continue;
+            }
+
+            if ($groupType === GroupType::REGION) {
+                $parentGroup = $group->getParentGroup();
+                if ($parentGroup == null) {
+                    $parentlessDepartments[] = $quap;
+                    continue;
+                }
+                $parentGroupId = $parentGroup->getId();
+
+                $nestedAnswer = $nestedDepartmentsMap[$parentGroupId] ?? null;
+                if ($nestedAnswer === null) {
+                    $nestedDepartmentsMap[$parentGroupId] = new HierarchyDTO(null, array(new HierarchyDTO($quap)));
+                    continue;
+                }
+
+                $nestedAnswer->addChild(new HierarchyDTO($quap));
+                continue;
+            }
+
+            $parentGroup = $group->getParentGroup();
+            if ($parentGroup == null) {
+                $parentlessDepartments[] = $quap;
+                continue;
+            }
+            $parentGroupId = $parentGroup->getId();
+
+            $cantonId = $group->getCantonId();
+            if ($cantonId === null) {
+                $parentlessDepartments[] = $quap;
+            }
+
+            $cantonalHierarchy = $nestedDepartmentsMap[$cantonId] ?? null;
+            if ($cantonalHierarchy === null) {
+                $parentlessDepartments[] = $quap;
+                continue;
+            }
+
+            $found = false;
+            foreach ($cantonalHierarchy->getChildren() as $regionalHierarchy) {
+                $regionalQuap = $regionalHierarchy->getParent();
+                if ($regionalQuap->getGroup()->getId() === $parentGroupId) {
+                    $regionalHierarchy->addChild(new HierarchyDTO($quap));
+                    $found = true;
+                    break;
+                }
```

Comment:
does this logic find regions which were created on line 324 where the the region itself is nested?